### PR TITLE
GCI: Replace BookBrainz jade template (oswinso)

### PR DESCRIPTION
**Pull request by oswinso**

### CODE REVIEW

File: src/server/templates/target.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		  <link rel="stylesheet" href='/stylesheets/react-virtualized.css' />
+		  <link rel="stylesheet"
+		    href='/stylesheets/react-virtualized-select.css' />
+		  <meta name='viewport' content='//react-virtualized.css' />
+		  ${favicon}
+		</head>
+
+		<body>
+		  <div id="target">${markup}</div>
+		  <script src='/js/bundle.js' />
```
Comment (Arthelon) [inline src/server/templates/target.js]: Both the starting and ending tags are [required for the script element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script).

File: src/server/templates/target.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		  ${page ?
```
Comment (Arthelon) [inline src/server/templates/target.js]: It'd be clearer to use a logical `&&` here instead of a ternary expression since the falsy condition is empty. Do the same for the other ternary expression below.

File: src/server/templates/target.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		    `<script id='page' type='application/json' >${page}</script>` : ''}
+		  ${props && script ?
+		    `<script id='props' type='application/json'> ${props}</script>
+			<script src='${script} />` : ''}
```
Comment (Arthelon) [inline src/server/templates/target.js]: Same issue as the previous script tag. Also try to keep indentation consistent.

File: src/server/templates/target.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	  <!DOCTYPE html>
+	  <html>
+		<head>
+		  <title>${title ? `'${title} – BookBrainz'` :
+		     'BookBrainz – The Open Book Database'}</title>
+		  <link rel="stylesheet" href='/stylesheets/style.css' />
+		  <link rel="stylesheet" href='/stylesheets/react-virtualized.css' />
+		  <link rel="stylesheet"
+		    href='/stylesheets/react-virtualized-select.css' />
+		  <meta name='viewport' content='//react-virtualized.css' />
```
Comment (Arthelon) [inline src/server/templates/target.js]: Please fix the `content` value for this meta tag. Your use of quotations is also inconsistent, please stick to single-quotes since it is enforced in the rest of the project.

File: src/server/templates/target.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import favicon from './favicon';
+
+
+export default ({title, markup, page, props, script}) => `
+	  <!DOCTYPE html>
+	  <html>
+		<head>
+		  <title>${title ? `'${title} – BookBrainz'` :
```
Comment (Arthelon) [inline src/server/templates/target.js]: Please remove the single quotes nested inside the inner template string.

File: src/server/templates/target.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+		  ${favicon}
+		</head>
+
+		<body>
+		  <div id="target">${markup}</div>
+		  <script src='/js/bundle.js' />
+		  ${page ?
+		    `<script id='page' type='application/json' >${page}</script>` : ''}
+		  ${props && script ?
+		    `<script id='props' type='application/json'> ${props}</script>
```
Comment (Arthelon) [inline src/server/templates/target.js]: unnecessary whitespace

File: src/server/templates/target.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+		    href='/stylesheets/react-virtualized-select.css' />
+		  <meta name='viewport' content='//react-virtualized.css' />
+		  ${favicon}
+		</head>
+
+		<body>
+		  <div id="target">${markup}</div>
+		  <script src='/js/bundle.js' />
+		  ${page ?
+		    `<script id='page' type='application/json' >${page}</script>` : ''}
```
Comment (Arthelon) [inline src/server/templates/target.js]: unnecessary whitespace after `type` attribute

File: src/server/templates/target.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+	  <html>
+		<head>
+		  <title>${title ? `'${title} – BookBrainz'` :
+		     'BookBrainz – The Open Book Database'}</title>
+		  <link rel="stylesheet" href='/stylesheets/style.css' />
+		  <link rel="stylesheet" href='/stylesheets/react-virtualized.css' />
+		  <link rel="stylesheet"
+		    href='/stylesheets/react-virtualized-select.css' />
+		  <meta name='viewport' content='//react-virtualized.css' />
+		  ${favicon}
```
Comment (Arthelon) [inline src/server/templates/target.js]: `favicon` is imported as a function that returns a template string but it is never invoked. I recommend combining `favicon.js` into `target.js` and adding it as a constant for the time being.

**Comment by oswinso** [comment]
@Arthelon Should I be using the custom express template engine here for the GCI task?

**Comment by oswinso** [comment]
@Arthelon Styling errors fixed with d07bf14

File: src/server/templates/favicon.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+export default () => `
```
Comment (Arthelon) [inline src/server/templates/favicon.js]: This file is unused and can be deleted.

File: src/server/templates/target.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
+const favicon = `
+	<link rel="apple-touch-icon" sizes="57x57"
```
Comment (Arthelon) [inline src/server/templates/target.js]: replace all double-quotes here with single quotes as per the previous review.

File: src/server/templates/target.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+	title,
+	markup,
+	page,
+	props,
+	script
+}) => `
+	<!DOCTYPE html>
+	<html>
+		<head>
+		<title>${title ? `${title} – BookBrainz` :
```
Comment (Arthelon) [inline src/server/templates/target.js]: indent the following 3 lines one level outwards.

File: src/server/templates/target.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+		  <link rel='stylesheet'
+		    href='/stylesheets/react-virtualized-select.css' />
+		  <meta name='viewport' content='width=device-width, initial-scale=1' />
+		  ${favicon}
+		</head>
+
+		<body>
+			<div id='target'>${markup}</div>
+			<script src='/js/bundle.js'></script>
+			${page && script ?
```
Comment (Arthelon) [inline src/server/templates/target.js]: The conditional logic here doesn't match the jade template. It should look something more like this:

Comment (Arthelon) [inline src/server/templates/target.js]: The conditional logic here doesn't match the jade template. It should look something more like this (you shouldn't need to use a ternary expression):

```
${page && ...}
${props && script && ...}
```

**Comment by oswinso** [comment]
@Arthelon Fixed again with d0de250

Sorry...

**Comment by Arthelon** [comment]
Looks good!
